gravitee-apim-console-webui/src/management/settings/identity-providers-ng/identity-providers-ng.component.ts

```diff
@@ -0,0 +1,187 @@
+import { gioTableFilterCollection } from '../../../shared/components/gio-table-wrapper/gio-table-wrapper.util';
+import { PortalSettings } from '../../../entities/portal/portalSettings';
+
+interface IdentityProviderForm {
+  forceLogin: FormGroup<{
+    enabled: FormControl<boolean>;
+  }>;
+  localLogin: FormGroup<{
+    enabled: FormControl<boolean>;
+  }>;
+}
+
+export interface ActivatedIdentityProvider {
+  identityProvider: string;
+}
+
+@Component({
+  selector: 'identity-providers-ng',
+  templateUrl: './identity-providers-ng.component.html',
+  styleUrls: ['./identity-providers-ng.component.scss'],
+})
+export class IdentityProvidersNgComponent implements OnInit {
+  identityProvidersForm: FormGroup;
+  settings: PortalSettings;
+  activatedIdentityProvider: ActivatedIdentityProvider[] = [];
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  public formInitialValues: unknown;
+  public identityProviderListTable: IdentityProvider[] = [];
+  public isLoadingData = true;
+  public displayedColumns = ['logo', 'id', 'name', 'description', 'actions'];
+  public filteredIdentityProviderSettingsTable: IdentityProvider[] = [];
+  public identityProviderUnpaginatedLength = 0;
+  public initialFilters: GioTableWrapperFilters = {
+    pagination: {
+      size: 10,
+      index: 1,
+    },
+    searchTerm: '',
+  };
+
+  constructor(
+    private readonly portalSettingsService: PortalSettingsService,
+    private readonly snackBarService: SnackBarService,
+    private readonly identityProviderService: IdentityProviderService,
+    private readonly matDialog: MatDialog,
+  ) {}
+
+  public ngOnInit() {
+    this.isLoadingData = true;
+
+    combineLatest([this.identityProviderService.list(), this.identityProviderService.listActivated(), this.portalSettingsService.get()])
+      .pipe(
+        tap(([identityProvider, activatedIdentityProvider, portalSettings]) => {
+          this.activatedIdentityProvider = activatedIdentityProvider.map((activated) => ({ identityProvider: activated.identityProvider }));
+          this.identityProviderListTable = identityProvider.map((identityProvider) => {
+            const matchedId = activatedIdentityProvider.find((item) => item.identityProvider === identityProvider.id);
+            return {
+              ...identityProvider,
+              isActivated: !!matchedId,
+              logo: `assets/logo_${identityProvider.type.toLowerCase()}-idp.svg`,
+            };
+          });
+          this.onPropertiesFiltersChanged(this.initialFilters);
+          this.settings = portalSettings;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => {
+        this.isLoadingData = false;
+        this.initIdentityProvidersForm();
+      });
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  onSubmit() {
+    this.portalSettingsService
+      .save({
```

Comment:
WDYT to get data before save ? 

```suggestion
     this.portalSettingsService
      .get().pipe(
      switchMap((settings) => {
          const updatedSettings = {
      ...settings,
      authentication: {
        ...settings.authentication,
        ...
      },
    };
    
    return this.environmentSettingsService.save(updatedSettings)
    
    })
    ....
```

Reply:
Done

Reply:
@ThibaudAV it changes a test a little bit
before I had 
`      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.env.baseURL}/settings`);
      expect(req.request.method).toEqual('PUT');
`
now I have 
`      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.env.baseURL}/settings`);
      expect(req.request.method).toEqual('GET');
`
to pass the test

Reply:
yes one get before and the the PUT. right ?

Reply:
Done